tests/petsc/assemble_matrix_parallel_01.cc

```diff
@@ -384,6 +381,12 @@ template <int dim>
 void
 LaplaceProblem<dim>::assemble_test()
 {
+  // This test requires PETSc to be configured with the option
+  // "--with-threadsafety" in case of debug builds of PETSc.
+  // For optimized builds, the above option is needed only in case
+  // users want PETSc to produce useful logs with "-log_view" runtime
+  // option
+#if !defined(PETSC_USE_DEBUG) || defined(PETSC_HAVE_THREADSAFETY)
```

Comment:
This will make the test fail if that flag is not set (because the output is different), right? If so, I would prefer to just abort the test.

What happens if `PETSC_USE_DEBUG` *is* defined?

Reply:
The test won't fail  since it will print zero as error

Reply:
With `PETSC_USE_DEBUG` petsc uses additional non threadsafe structures to track entry and exit points of functions

Reply:
Oh, I see now -- the correct output is 0. Yes, this will work then.

Is `PETSC_USE_DEBUG` the default?

Reply:
sadly yes

Reply:
Ah, bummer :-(